adapters/adnuntius/adnuntius.go

```diff
@@ -0,0 +1,300 @@
+			ortbUserId := ortbRequest.User.ID
+			if ortbUserId != "" {
+				adnuntiusRequest.MetaData.Usi = ortbRequest.User.ID
+			}
+		}
+
+		adnJson, err := json.Marshal(adnuntiusRequest)
+		if err != nil {
+			return nil, err
+		}
+
+		requestData = append(requestData, &adapters.RequestData{
+			Method:  http.MethodPost,
+			Uri:     endpoint,
+			Body:    adnJson,
+			Headers: headers,
+		})
+
+	}
+
+	return requestData, nil
+}
+
+func (a *adapter) MakeBids(request *openrtb2.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if response.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	if response.StatusCode == http.StatusBadRequest {
+		return nil, []error{&errortypes.BadInput{
+			Message: fmt.Sprintf("Unknown status code: %d.", response.StatusCode),
+		}}
+	}
+
+	if response.StatusCode != http.StatusOK {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Unknown status code: %d.", response.StatusCode),
+		}}
+	}
+
+	var adnResponse AdnResponse
+	if err := json.Unmarshal(response.Body, &adnResponse); err != nil {
+		return nil, []error{err}
+	}
+
+	bidResponse, bidErr := generateBidResponse(&adnResponse, request)
+	if bidErr != nil {
+		return nil, bidErr
+	}
+
+	return bidResponse, nil
+}
+
+func getGDPR(request *openrtb2.BidRequest) (string, string, error) {
+	gdpr := ""
+	var extRegs openrtb_ext.ExtRegs
+	if request.Regs != nil {
+		if err := json.Unmarshal(request.Regs.Ext, &extRegs); err != nil {
+			return "", "", fmt.Errorf("failed to parse ExtRegs in Yieldlab GDPR check: %v", err)
+		}
+		if extRegs.GDPR != nil && (*extRegs.GDPR == 0 || *extRegs.GDPR == 1) {
+			gdpr = strconv.Itoa(int(*extRegs.GDPR))
+		}
+	}
+
+	consent := ""
+	if request.User != nil && request.User.Ext != nil {
+		var extUser openrtb_ext.ExtUser
+		if err := json.Unmarshal(request.User.Ext, &extUser); err != nil {
+			return "", "", fmt.Errorf("failed to parse ExtUser in Yieldlab GDPR check: %v", err)
+		}
+		consent = extUser.Consent
+	}
+
+	return gdpr, consent, nil
+}
+
+func generateBidResponse(adnResponse *AdnResponse, request *openrtb2.BidRequest) (*adapters.BidderResponse, []error) {
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(len(adnResponse.AdUnits))
+	currency := bidResponse.Currency
+
+	for i, adunit := range adnResponse.AdUnits {
+
+		var bid openrtb2.Bid
+		if len(adunit.Ads) > 0 {
+			ad := adunit.Ads[0]
+
+			currency = ad.Bid.Currency
+
+			creativeWidth, Werr := strconv.ParseInt(ad.CreativeWidth, 10, 64)
+			if Werr != nil {
+				return nil, []error{Werr}
+			}
+
+			creativeHeight, Herr := strconv.ParseInt(ad.CreativeHeight, 10, 64)
+			if Herr != nil {
+				return nil, []error{Herr}
+			}
+
+			adDomain := []string{}
+			for _, url := range ad.DestinationUrls {
+				domainArray := strings.Split(url, "/")
+				domain := strings.Replace(domainArray[2], "www.", "", -1)
+				adDomain = append(adDomain, domain)
```

Comment:
Are the destination URLs always going to start with `http(s)://`?
Nitpick: maybe split on `//` and grab `domainArray[1]` instead? IMO that seems a little more straightforward.

Reply:
Yeah, that seems like a good idea but I got the impression that i was only allowed to send the top level domain in that field? :D If not I can just stop what that stuff is doing and just pass the entire url. And yes, we force the url to have http(s)://